### DISCUSSION TITLE
Build simulator project on OSX - delay_ms is undefined

### DISCUSSION THREAD

**Discussion by mgussekloo** [joeycastillo/Sensor-Watch#88] [Q&A] [answered]
Hi, I'm trying to build the sample project on a Macbook Pro M1. I'm trying to run the simulator, as I haven't received my watch yet. I've installed the GNU Toolchain for ARM Embedded Processors, and Emscripten. Running 'emmake make' in the starter-project folder gives me the error below. Does anyone know how to fix this?

➜  starter-project git:(main) emmake make             
make: make
git submodule update --init
HTML build/watch.html
error: undefined symbol: delay_ms (referenced by top-level compiled C/C++ code)
warning: Link with `-sLLD_REPORT_UNDEFINED` to get more information on undefined symbols
warning: To disable errors for undefined symbols use `-sERROR_ON_UNDEFINED_SYMBOLS=0`
warning: _delay_ms may need to be added to EXPORTED_FUNCTIONS if it arrives from a system library
Error: Aborting compilation due to previous errors
emcc: error: '/opt/homebrew/opt/node/bin/node /opt/homebrew/Cellar/emscripten/3.1.23/libexec/src/compiler.js /var/folders/xh/8_2nml715d5053dhdsy4j49h0000gn/T/tmp1fjooxwv.json' failed (returned 1)

Update: Haven't gotten it to work. This is what I did so far: installed the toolchain via this [homebrew package](https://github.com/osx-cross/homebrew-arm). Running a "make clean" and "emmake make clean", I was able to run "make" fine. Running "emmake make" still results in "undefined symbol: delay_ms". If I comment the delay_ms lines from Starter-project's app.c, it compiles.

**Comment by wryun** [comment]
It fails for me too. However, it works to build 'movement' - I've made a PR with some more instructions on getting started after I, err, got started yesterday:

https://github.com/wryun/Sensor-Watch/tree/readme-movement#using-the-movement-framework

**Comment by wryun** [comment]
(suspect that the simulator has only been tested with movement, and is missing some HAL bit)

**Comment by wryun** [comment, marked as answer]
https://github.com/joeycastillo/Sensor-Watch/pull/93

**Comment by mgussekloo** [comment]
Thanks wryun, this fixes the issue!
